blueflood-core/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/RollupHandler.java

```diff
@@ -206,4 +268,28 @@ protected MetricData getHistogramsByGranularity(String tenantId,
 
         return data;
     }
+
+    private class UnitsGetter implements Callable {
+        private String tenantId;
+        private String metricName;
+
+        public UnitsGetter(String tenantId, String metricName) {
+            this.tenantId = tenantId;
+            this.metricName = metricName;
+        }
+
+        public String call() {
+            String unit = null;
+            try {
+                List<SearchResult> results = discoveryHandler.search(tenantId, metricName);
+                for (SearchResult res : results) {
+                    unit = res.getUnit();
+                    break;
+                }
+            } catch (Exception ex) {
+                log.info(ex.getMessage());
```

Comment:
May be we should just let the callable throw an exception and catch it in the block around `.get()`